pkg/network/ebpf/c/protocols/redis/decoding.h

```diff
@@ -2,19 +2,290 @@
 #define __REDIS_DECODING_H
 
 #include "protocols/redis/decoding-maps.h"
+#include "protocols/helpers/pktbuf.h"
+
+#define BLK_SIZE (16)
+PKTBUF_READ_INTO_BUFFER(redis_bulk, MAX_KEY_LEN, BLK_SIZE)
+
+// Read a CRLF terminator from the packet buffer. The terminator is expected to be in the format: \r\n.
+// The function returns true if the terminator was successfully read, or false if the terminator could not be read.
+static __always_inline bool read_crlf(pktbuf_t pkt) {
+    char terminator[RESP_FIELD_TERMINATOR_LEN];
+    if (pktbuf_load_bytes_from_current_offset(pkt, terminator, RESP_FIELD_TERMINATOR_LEN) < 0) {
+        return false;
+    }
+    pktbuf_advance(pkt, RESP_FIELD_TERMINATOR_LEN);
+    return terminator[0] == RESP_TERMINATOR_1 && terminator[1] == RESP_TERMINATOR_2;
+}
+
+// Read an array message from the packet buffer. The array message is expected to be in the format:
+// *<param_count>\r\n<param1>\r\n<param2>\r\n...
+// where <param_count> is the number of parameters in the array, and <param1>, <param2>, etc. are the parameters themselves.
+// The function returns the number of parameters in the array, or 0 if the array message could not be read.
+static __always_inline u32 read_array_message_param_count(pktbuf_t pkt) {
+    // Verify RESP array prefix
+    char first_byte;
+    if (pktbuf_load_bytes_from_current_offset(pkt, &first_byte, sizeof(first_byte)) < 0 || first_byte != RESP_ARRAY_PREFIX) {
+        return 0;
+    }
+    pktbuf_advance(pkt, sizeof(first_byte));
+
+    // Read parameter count
+    char param_count;
+    // Assuming single-digit param count, as currently we don't need more.
+    if (pktbuf_load_bytes_from_current_offset(pkt, &param_count, 1) < 0) {
+        return 0;
+    }
+    pktbuf_advance(pkt, sizeof(param_count));
+
+    if (param_count < '0' || param_count > '9') {
+        return 0;
+    }
+
+    if (!read_crlf(pkt)) {
+        return 0;
+    }
+
+    return param_count - '0';
+}
+
+static __always_inline u16 get_key_len(pktbuf_t pkt) {
+    u32 current_offset = pktbuf_data_offset(pkt);
+    const u32 data_end = pktbuf_data_end(pkt);
+
+    char bulk_prefix;
+    // Verify we can read the RESP bulk prefix.
+    if (current_offset + sizeof(bulk_prefix) > data_end) {
+        return 0;
+    }
+    if (pktbuf_load_bytes(pkt, current_offset, &bulk_prefix, sizeof(bulk_prefix)) < 0 || bulk_prefix != RESP_BULK_PREFIX) {
+        return 0;
+    }
+    current_offset++;
+
+    // Read key length (up to 3 digits)
+    char key_size_bytes[3] = {};
+    if (current_offset + sizeof(key_size_bytes) > data_end) {
+        return 0;
+    }
+    if (pktbuf_load_bytes(pkt, current_offset, key_size_bytes, sizeof(key_size_bytes)) < 0) {
+        return 0;
+    }
+
+    u16 key_size = 0;
+    u32 digits_read = 0;
+    // The key length is a decimal number, so we need to convert it from ASCII to an integer.
+    #pragma unroll (3)
```

Comment:
Why 3? I would suggest giving descriptive names to all of the numbers by using constants.